src/base/static/components/atoms/input.js

```diff
@@ -23,6 +23,28 @@ CheckboxInput.propTypes = {
   onChange: PropTypes.func.isRequired,
 };
 
+const NumberInput = props => {
+  return (
+    <input
+      className={props.className}
```

Comment:
I feel like this isn't right and I'm still missing something about styled components. We shouldn't have to pass the emotion className through here, right?

Reply:
No, we shouldn't be passing class names through our props. It's an anti-pattern.

Instead, try this:

```javascript
const BasicNumberInput = props => (<input type="number" name={props.name} value={props.value} .... />)
const NumberInput = styled(BasicNumberInput)(props => ({  
  // default styles:
  border: "0.25em solid #a3c7d9",
  padding: "0.5em",
  boxSizing: "border-box",
  width: "100%",
  // etc.
  
  // style overrides based on props:
  backgroundColor: props.hasAutofill ? "#ffffe0" : "#ffffff",
  // ... etc


}))
```

Does that make sense?

Reply:
For some reason I still can't get this pattern to work. I might be missing something basic, but the emotion docs suggest passing through the classname here:

https://emotion.sh/docs/styled#styling-any-component

It seems like emotion uses classes behind the scenes?